frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import * as fuzzy from 'fuzzysearch';
+
+import { Firehose, LoadingInline, Dropdown, ResourceName, ResourceIcon } from '.';
+
+/* Component StorageClassDropdown - creates a dropdown list of storage classes */
+
+class StorageClassDropdown_ extends React.Component<StorageClassDropdownProps, StorageClassDropdownState> {
+  readonly state: StorageClassDropdownState = {
+    items: {},
+    name: this.props.name,
+    selectedKey: null,
+    title: <LoadingInline />,
+    defaultClass: null,
+  };
+
+  componentWillMount() {
+    this.componentWillReceiveProps(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { loaded, loadError, resources } = nextProps;
+
+    if (loadError) {
+      this.setState({
+        title: (
+          <div className="cos-error-title">Error Loading {nextProps.desc}</div>
+        ),
+      });
+      return;
+    }
+    if (!loaded) {
+      return;
+    }
+
+    const state = {
+      items: {},
+      title: {},
+      defaultClass: '',
+    };
+    const unorderedItems = {};
+    const noStorageClass = 'No storage class';
+    _.each(resources, ({ data }, kindLabel) => {
```

Comment:
Isn't there always one resource? I'm not sure this `_.each` is needed. I've just give the prop name `storageClasses` and then iterator over `storageClasses.data`